spec/javascripts/components/search-with-autocomplete-spec.js

```diff
@@ -215,4 +215,69 @@ describe('Search with autocomplete component', () => {
     expect(formData.get('q')).toEqual('updated value')
     expect(submitSpy).toHaveBeenCalled()
   })
+
+  describe('analytics data attributes', () => {
+    it('sets data attributes on the input when suggestions are returned', (done) => {
+      const input = fixture.querySelector('input')
+
+      stubSuccessfulFetch([
+        'my favourite song is red',
+        'my favourite song is karma',
+        'my favourite song is death by a thousand cuts'
+      ])
+      performInput(input, 'my favourite song is', () => {
+        expect(input.getAttribute('data-autocomplete-trigger-input')).toEqual('my favourite song is')
+        expect(input.getAttribute('data-autocomplete-suggestions')).toEqual(
+          'my favourite song is red|' +
+          'my favourite song is karma|' +
+          'my favourite song is death by a thousand cuts'
+        )
+        expect(input.getAttribute('data-autocomplete-suggestions-count')).toEqual('3')
+        expect(input.getAttribute('data-autocomplete-accepted')).toEqual('false')
+        done()
+      })
+    })
+  })
+
+  it('limits the number of suggestions included in the data to 5', (done) => {
+    const input = fixture.querySelector('input')
+
+    stubSuccessfulFetch([
+      'my favourite album is red',
+      'my favourite album is midnights',
+      'my favourite album is lover',
+      'my favourite album is folklore',
+      'my favourite album is reputation',
+      'my favourite album is 1989'
+    ])
```

Comment:
Had not realised you were a swiftie 😁

Reply:
Oh you'll enjoy the Finder Frontend search tests 😂 I've almost crowded out @sihugh's Harry Potter references!